src/containers/Application/ApplicationOverview.tsx

```diff
@@ -74,7 +76,7 @@ const ApplicationOverview: React.FC = () => {
     if (isApplicationLoaded && elementsState && responsesByCode) {
       revalidateAndUpdate().then(() => setIsRevalidated(true))
     }
-  }, [responsesByCode, elementsState])
+  }, [responsesByCode, elementsState, appStatus])
```

Comment:
i think this `appStatus` can be removed now along with the `console.log` commented on the top.

Reply:
No, the lack of this `appStatus` is the main thing that was causing the problem! 😄

Because the first time `appStatus` comes back `undefined`, in which case `isRevalidated` never gets set to `true`, and the application doesn't load in the UI. And so we need to make sure this `useEffect` re-runs when `appStatus` is eventually updated with it's proper value (in which case it skips revalidation and sets `isRevalidated` to `true`).

Does that make sense? It's been doing my head in for a while.